datafusion/core/src/physical_plan/aggregates/group_values/mod.rs

```diff
@@ -42,6 +43,9 @@ pub trait GroupValues: Send {
 
     /// Emits the group values
     fn emit(&mut self, emit_to: EmitTo) -> Result<Vec<ArrayRef>>;
+
+    /// clear the contents and shrink the capacity to free up memory usage
```

Comment:
The intention of passing `RecordBatch` is to use other information than number of rows for reserving the new capacity such as data (binary) capacity. But it will be separate future PRs.

Reply:
Oh cool, perhaps its worth pointing out why `RecordBatch` is required in the comments too.

Reply:
Updated the comment